pep/app/services/current-user.ts

```diff
@@ -199,10 +203,9 @@ export default class CurrentUserService extends Service {
      * @memberof CurrentUserService
      */
     addPreferenceDocument(key: PreferenceDocumentsKey, documentId: string) {
```

Comment:
updated these methods to use pure immutable update logic, I forgot to mention that requirement on the other PR passes.